Mathlib/Analysis/SpecialFunctions/PolarCoord.lean

```diff
@@ -169,7 +169,8 @@ protected theorem polarCoord_apply (a : ℂ) :
   rfl
 
 protected theorem polarCoord_source :
-    Complex.polarCoord.source = {a | 0 < a.re} ∪ {a | a.im ≠ 0} := by simp [Complex.polarCoord]
+    Complex.polarCoord.source = Complex.slitPlane := by
```

Comment:
Could you change the definition of `polarCoord` to make sure that this becomes definitional? Better use `slitPlane` everywhere it's natural!

Reply:
The problem here is that `Complex.polarCoord` is defined via `polarCoord`, whose source is a subset of `ℝ × ℝ` and not `ℂ`. So I think to get this definitional equality, one would have to define `Complex.polarCoord` directly. I'll try this out.

Reply:
After playing around for a while, I don't think there is a good way of making this work. Note that the original version also was not `rfl`: expanding the `simp` gives
```lean
  simp only [Complex.polarCoord, PartialHomeomorph.trans_toPartialEquiv, PartialEquiv.trans_source,
    Homeomorph.toPartialHomeomorph_source, PartialHomeomorph.toFun_eq_coe,
    Homeomorph.toPartialHomeomorph_apply, ContinuousLinearEquiv.coe_toHomeomorph, polarCoord_source,
    ne_eq, Set.preimage_union, Set.preimage_setOf_eq, equivRealProdClm_apply, Set.univ_inter]
```
Really defining `Complex.polarCoord` from scratch could work, but I don't think getting the defeq for the source is worth the code duplication.

Reply:
Fixed by using `Homeomorph.transPartialHomeomorph` instead of converting a `Homeomorph` to a `PartialHomeomorph` first.